data/bash-completion/fwupdtool

```diff
@@ -77,6 +77,15 @@ _show_firmware_types()
 	COMPREPLY+=( $(compgen -W "${firmware_types}" -- "$cur") )
 }
 
+_show_device_ids()
+{
+	if ! command -v jq &> /dev/null; then
+		return 0
+	fi
+	local description
+	description="$(command jq '.Devices | .[] | .DeviceId' /var/cache/fwupd/devices.json 2>/dev/null)"
```

Comment:
I don't like hardcoding this path; can we perhaps instead fill in the `/var/cache/fwupd` from meson?

Reply:
> I don't like hardcoding this path; can we perhaps instead fill in the `/var/cache/fwupd` from meson?

The other way around is to implement the load from json in `fwupdtool` directly (https://github.com/fwupd/fwupd/pull/4493#discussion_r896652875).

Reply:
Either way is fine to me.  If you keep it this way just use `configure_file` to populate the path. If you switch to `fwupdtool` to parse it just make sure that it's fast path (don't load plugins and engine and stuff).

Reply:
> Either way is fine to me. If you keep it this way just use `configure_file` to populate the path. If you switch to `fwupdtool` to parse it just make sure that it's fast path (don't load plugins and engine and stuff).

I will go for the `configure_file` and then switch back to `fwupdtool` in a later PR as it takes lots of changes.